integration/testrunner/db.go

```diff
@@ -82,34 +104,44 @@ func makeBigQueryClient(ctx context.Context, tb testing.TB, projectID string) *b
 	return client
 }
 
-// This calls the database to check that an audit log exists. It uses the retries that are specified in the Config
-// file. This method assumes that only a single audit log will match, which constitutes success.
-func queryIfAuditLogExistsWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string) {
-	tb.Helper()
-
-	queryIfAuditLogsExistWithRetries(ctx, tb, bqQuery, cfg, testName, 1)
-}
-
 // This calls the database to check that an audit log exists. It uses the retries that are specified in the Config
 // file. This method allows for specifying how many logs we expect to match, in order to handle streaming use cases.
-func queryIfAuditLogsExistWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string, expectedNum int64) {
+func queryAuditLogsWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string) []*logpb.LogEntry {
 	tb.Helper()
-
+	var logEntries []*logpb.LogEntry
 	b := retry.NewExponential(cfg.LogRoutingWait)
 	if err := retry.Do(ctx, retry.WithMaxRetries(cfg.MaxDBQueryTries, b), func(ctx context.Context) error {
-		found, err := queryIfAuditLogExists(ctx, tb, bqQuery, expectedNum)
-		if found {
+		results, err := queryAuditLogs(ctx, tb, bqQuery)
+		if results != nil {
 			// Early exit retry if queried log already found.
+			logEntries = results
 			return nil
 		}
 
 		tb.Log("Matching entry not found, retrying...")
 
 		if err != nil {
-			tb.Logf("Query error: %v.", err)
+			return retry.RetryableError(fmt.Errorf("query error: %w", err))
```

Comment:
I don't think this should retryable error. The only retryable error should be "we got results successfully but they are not what we want"

Reply:
I think some cases are retryable, like `failed to run query`, `failed to wait for query`, `failed to read job`, should we call `t.Fatal` for non-retryable error and return error for retryable error?